app/assets/javascripts/discourse/controllers/full-page-search.js.es6

```diff
@@ -73,32 +75,31 @@ export default Ember.Controller.extend({
 
   @computed('q')
   noSortQ(q) {
-    if (q) {
-      SortOrders.forEach((order) => {
-        if (q.indexOf(order.term) > -1){
-          q = q.replace(order.term, "");
-          q = q.trim();
-        }
-      });
-    }
+    q = this.cleanTerm(q);
     return escapeExpression(q);
   },
 
   _searchOnSortChange: true,
 
   setSearchTerm(term) {
     this._searchOnSortChange = false;
+    term = this.cleanTerm(term);
+    this._searchOnSortChange = true;
+    this.set('searchTerm', term);
+  },
+
+  cleanTerm(term) {
     if (term) {
       SortOrders.forEach(order => {
-        if (term.indexOf(order.term) > -1){
+        let matches = term.match(new RegExp(`${order.term}\\b`));
+        if (matches) {
           this.set('sortOrder', order.id);
           term = term.replace(order.term, "");
```

Comment:
I think you'll need to replace using the regexp too. Otherwise it could replace the first occurrence which is smaller even if it found a later one.